web/packages/teleterm/src/services/config/getKeyboardShortcutSchema.ts

```diff
@@ -0,0 +1,175 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { z } from 'zod';
+
+import { Platform } from 'teleterm/mainProcess/types';
+
+const VALID_SHORTCUT_MESSAGE =
+  'Valid shortcut can contain multiple modifiers (like "Shift") and a single key code (like "A" or "Tab"), combined by the "+" character.';
+
+export function invalidKeyCodeIssue(wrongKeyCode: string): z.IssueData {
+  return {
+    code: z.ZodIssueCode.custom,
+    message: `"${wrongKeyCode}" cannot be used as a key code. ${VALID_SHORTCUT_MESSAGE}`,
+  };
+}
+
+export function invalidModifierIssue(wrongModifier: string): z.IssueData {
+  return {
+    code: z.ZodIssueCode.custom,
+    message: `"${wrongModifier}" cannot be used as a modifier. ${VALID_SHORTCUT_MESSAGE}`,
+  };
+}
+
+export function duplicatedModifierIssue(): z.IssueData {
+  return {
+    code: z.ZodIssueCode.custom,
+    message: `Duplicated modifier found. ${VALID_SHORTCUT_MESSAGE}`,
+  };
+}
+
+export function getKeyboardShortcutSchema(platform: Platform) {
+  const allowedModifiers = getAllowedModifiers(platform);
+
+  return z
+    .string()
+    .transform(s => s.split('+'))
```

Comment:
What about whitespace? I'm thinking we could trim the whitespace on the original string and then split by `<optional whitespace>+<optional whitespace>` to not punish people for setting `Shift + A` as the accelerator.

Reply:
Good idea. I think we can just replace all spaces with `''` and then split by "+" as usual.

Reply:
> Good idea. I think we can just replace all spaces with `''` and then split by "+" as usual.

I think it's better to be more intentional/specific in cases like this one. If we remove all whitespaces, then something like "A lt+F 4" will be a valid value.

Reply:
Right, I will change it.